src/Compilers/CSharp/Portable/Binder/LookupResult.cs

```diff
@@ -320,5 +320,15 @@ internal void Free()
                 _pool.Free(this);
             }
         }
+
+        internal static Disposer GetInstance(out LookupResult instance)
+            => new Disposer(instance = GetInstance());
+
+        internal readonly ref struct Disposer
```

Comment:
I recall some PRs with similar proposed design change (using disposal instead of manual freeing) that were previously rejected.
Here's one that I found and the relevant comment from Andy:
"I believe we explicitly decided not to do this because pooled objects do not satisfy the IDisposable constraint because they cannot be disposed multiple times."
[https://github.com/dotnet/roslyn/pull/37543#pullrequestreview-268178649](https://github.com/dotnet/roslyn/pull/37543#pullrequestreview-268178649)

Reply:
More recent discussion: https://github.com/dotnet/roslyn/issues/50957

I don't have a strong opinion here, but to resound to the quotes. 

"It introduces a virtual call instead of a direct call for freeing." This is a ref struct.
"It introduces a try-finally, which is very expensive in the CLR." That's interesting. try-finally is already being used a lot and sometimes only for early returns. Pretty sure if I were using try-finally it was acceptable, no?

Reply:
In any case, I will revert.